src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/securityrule/sdk/AzureNetworkSecurityGroupOperationSDKTest.java

```diff
@@ -356,65 +383,111 @@ public void testBuildSecurityRuleInstanceSuccessfully() {
 
         Assert.assertTrue(networkSecurityRules.contains(securityRuleInstance));
     }
 
+    // test case: When calling the buildNetworkSecurityGroupId method, it must
+    // verify that the resource ID was assembled correctly.
+    @Test
+    public void testBuildNetworkSecurityGroupIdSuccessfully() throws Exception {
+        // set up
+        Azure azure = PowerMockito.mock(Azure.class);
+        String subscriptionId = AzureTestUtils.DEFAULT_SUBSCRIPTION_ID;
+        String networkSecurityGroupName = AzureTestUtils.RESOURCE_NAME;
+
+        String resourceGroupName = AzureTestUtils.DEFAULT_RESOURCE_GROUP_NAME;
+        PowerMockito.mockStatic(AzureGeneralUtil.class);
+        PowerMockito.doReturn(resourceGroupName).when(AzureGeneralUtil.class, "selectResourceGroupName",
+                Mockito.eq(azure), Mockito.eq(networkSecurityGroupName), Mockito.eq(this.defaultResourceGroupName));
+
+        String expected = createNetworkSecurityGroupId();
+
+        // exercise
+        String resourceId = this.operation.buildNetworkSecurityGroupId(azure, subscriptionId, networkSecurityGroupName);
+
+        // verify
+        PowerMockito.verifyStatic(AzureGeneralUtil.class, Mockito.times(TestUtils.RUN_ONCE));
+        AzureGeneralUtil.selectResourceGroupName(Mockito.eq(azure), Mockito.eq(networkSecurityGroupName),
+                Mockito.eq(this.defaultResourceGroupName));
+
+        Assert.assertEquals(expected, resourceId);
+    }
+
+    private SecurityRuleInstance createSecurityRuleInstance(String networkSecurityGroupName, String securityRuleName) {
+        String instanceId = SecurityRuleIdContext.buildInstanceId(networkSecurityGroupName, securityRuleName);
+        String cidr = TestUtils.DEFAULT_CIDR;
+        int portFrom = DEFAULT_PORT_FROM;
+        int portTo = DEFAULT_PORT_TO;
+        SecurityRule.Direction direction = SecurityRule.Direction.IN;
+        SecurityRule.EtherType etherType = SecurityRule.EtherType.IPv4;
+        SecurityRule.Protocol protocol = SecurityRule.Protocol.ANY;
+        return new SecurityRuleInstance(instanceId, direction, portFrom, portTo, cidr, etherType, protocol);
+    }
+
     private NetworkSecurityRule createSecurityRule(String name, int priority) {
         NetworkSecurityRule networkSecurityRule = Mockito.mock(NetworkSecurityRule.class);
         Mockito.when(networkSecurityRule.priority()).thenReturn(priority);
         Mockito.when(networkSecurityRule.name()).thenReturn(name);
         return networkSecurityRule;
     }
 
-    private void mockAzureClient() throws UnauthenticatedUserException {
-        PowerMockito.mockStatic(AzureClientCacheManager.class);
-        PowerMockito.when(AzureClientCacheManager.getAzure(Mockito.eq(this.azureUser)))
-                .thenReturn(azure);
+    private String createNetworkSecurityGroupId() {
+        String networkSecurityGroupIdFormat = "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/networkSecurityGroups/%s";
```

Comment:
The AzureResourceIdBuilder has this value as a constant.

Reply:
The visibility of the constants defined in the AzureResourceIdBuilder class does not allow them to be used by other classes, except for the test class itself.

Reply:
Can you see a problem about open the constant visibility?